### PR TITLE
Revert "Fix enable LLVM for JIT compilation in CMake"

### PR DESCRIPTION
**Pull request by tavplubix**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Reverts ClickHouse/ClickHouse#35683

Because it breaks `02152_http_external_tables_memory_tracking`

cc: @kitaisreal

### CODE REVIEW

**Comment by kitaisreal** [comment]
@tavplubix previous pull request just return code in cmake that was before, it must not break anything. Better to find where the issue is without reverting.

**Comment by tavplubix** [comment]
> it must not break anything

But looks like it breaks (I don't know how). 

> Better to find where the issue is without reverting.

It would be better to find the issue before merging. Currently functional tests in master are broken. Usually there are two ways: find the issue and fix it or find the PR that has broken tests and revert it. I'm waiting for tests to finish and going to merge this PR if tests will pass because currently we don't have a better fix.

**Review by kitaisreal — APPROVED** [review]

**Comment by tavplubix** [comment]
Functional tests are ok

**Pull request merged**
